lib/page/home/home_view_model.dart

```diff
@@ -53,7 +68,7 @@ class HomeViewModel extends StateNotifier<HomeState> {
       state = const HomeState.apiLoadingSuccess();
       _surveysPageNumber++;
     } else {
-      state = HomeState.error((result as Failed).exception);
```

Comment:
I think we should try to display the root cause of the error first before showing a generic error message.